File: frontend/src/api/appLock.ts
import { buildApiUrl, parseJsonResponse } from "./client";

export type AppLockStatus = { enabled: boolean; unlocked: boolean; message?: string };

async function request(path: string, body?: Record<string, string>): Promise<AppLockStatus> {
  const response = await fetch(await buildApiUrl(`/api/app-lock${path}`), {
    method: body ? "POST" : "GET",
    headers: body ? { "Content-Type": "application/json" } : undefined,
    body: body ? JSON.stringify(body) : undefined,
  });
  return parseJsonResponse<AppLockStatus>(response);
}

export const getAppLockStatus = () => request("/status");
export const enableAppLock = (password: string, confirmPassword: string) => request("/enable", { password, confirm_password: confirmPassword });
export const unlockApp = (password: string) => request("/unlock", { password });
export const changeAppLockPassword = (currentPassword: string, newPassword: string, confirmPassword: string) => request("/change-password", { current_password: currentPassword, new_password: newPassword, confirm_password: confirmPassword });
export const disableAppLock = (currentPassword: string) => request("/disable", { current_password: currentPassword });
export const lockApp = () => request("/lock", {});

File: backend/app/schemas/app_lock.py
from pydantic import BaseModel


class PasswordPair(BaseModel):
    password: str
    confirm_password: str


class UnlockRequest(BaseModel):
    password: str


class ChangePasswordRequest(BaseModel):
    current_password: str
    new_password: str
    confirm_password: str


class DisableRequest(BaseModel):
    current_password: str


class AppLockStatus(BaseModel):
    enabled: bool
    unlocked: bool


class AppLockResult(AppLockStatus):
    message: str
